python/dgl/transforms/functional.py

```diff
@@ -3856,4 +3857,133 @@ def double_radius_node_labeling(g, src, dst):
 
     return F.tensor(z, F.int64)
 
+def shortest_dist(g, root=None, directed=True, return_paths=False):
+    r"""Compute shortest distance and paths on the given graph.
+
+    This function computes the shortest distance between all pairs of nodes
+    or between all nodes and a specified root node. Corresponding shortest
+    paths can be returned and represented with edge IDs. Note that only
+    unweighted cases are allowed.
+
+    Parameters
+    ----------
+    g : DGLGraph
+        The input graph. Must be homogeneous.
+    root : int, optional
+        The root node ID if the graph is provided with a root node.
+        Default: None. If a root node ID is provided, the function returns
+        single-source shortest paths. Otherwise, it returns all-pairs
+        shortest paths.
+    directed: bool, optional
+        A flag to indicate whether the graph is considered as directed.
+        Default: True. If False, reverse edges are considered for constructing
+        shortest paths.
+    return_paths: bool, optional
+        A flag to indicate whether to return paths. Default: False. If True,
+        return shortest paths represented with edge IDs.
+
+    Returns
+    -------
+    dist: Tensor
+        The shortest distance tensor of shape :math:`(N, N)` if root is None,
+        where :math:`N` is the number of nodes in the input graph. Or the
+        shortest distance tensor of shape :math:`(N,)` if a root node is
+        specified. The distance of unreachable pairs are filled with -1.
+    paths: Tensor, optional
+        Return the shortest paths tensor of shape :math:`(N, N, max_len)` if
+        root is None, or tensor of shape :math:`(N, max_len)` when a root node
+        is specified, where ``max_len`` is the length of the longest paths.
+        Each path vector consists of edge IDs padded with -1 in the end. The
+        paths tensor is only returned if ``return_paths`` is True.
+
+    Example
+    -------
+    >>> import dgl
+
+    >>> g = dgl.graph(([0, 1, 3], [1, 2, 1]))
+    >>> dgl.shortest_dist(g, root=0, directed=True)
+    tensor([ 0,  1,  2, -1])
+    >>> dgl.shortest_dist(g, root=None, directed=False, return_paths=True)
```

Comment:
```suggestion
    >>> dist, path = dgl.shortest_dist(g, root=None, directed=False, return_paths=True)
```
and then print `dist` and `path` for a better presentation.